server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -270,6 +270,14 @@ public ObjectMapper getObjectMapper(String name) {
         return mapperService.getObjectMapper(name);
     }
 
+    public boolean isMetadataField(String field) {
+        return mapperService.isMetadataField(field);
+    }
+
+    public Set<String> sourcePath(String fullName) {
```

Comment:
One nit: maybe this should be `sourcePaths`, given that it can return more than one?

Reply:
yes, in my iterations I did both, I went for singular for consistency with the name of the method in MapperService. I don't know what is more important between using the same name or using the correct name here. Maybe we should rename both as a followup?

Reply:
+1 for a followup